src/aave-v3/SupplyVault.sol

```diff
@@ -126,37 +122,40 @@ contract SupplyVault is ISupplyVault, SupplyVaultBase {
         view
         returns (address[] memory rewardTokens, uint256[] memory unclaimedAmounts)
     {
-        uint256 supply = totalSupply();
-        if (supply > 0) {
-            uint256[] memory claimableAmounts;
+        uint256[] memory claimableAmounts;
 
-            {
-                address[] memory poolTokens = new address[](1);
-                poolTokens[0] = poolToken;
+        {
+            address[] memory poolTokens = new address[](1);
+            poolTokens[0] = poolToken;
 
-                (rewardTokens, claimableAmounts) = rewardsManager.getAllUserRewards(
-                    poolTokens,
-                    address(this)
-                );
-            }
+            (rewardTokens, claimableAmounts) = rewardsManager.getAllUserRewards(
+                poolTokens,
+                address(this)
+            );
+        }
+
+        unclaimedAmounts = new uint256[](claimableAmounts.length);
+        uint256 supply = totalSupply();
 
-            for (uint256 i; i < rewardTokens.length; ) {
-                address rewardToken = rewardTokens[i];
-                UserRewardsData memory userRewardsData = userRewards[rewardToken][_user];
+        for (uint256 i; i < rewardTokens.length; ++i) {
+            address rewardToken = rewardTokens[i];
+            UserRewardsData memory userRewardsData = userRewards[rewardToken][_user];
 
+            if (supply > 0)
                 unclaimedAmounts[i] =
                     userRewardsData.unclaimed +
                     balanceOf(_user).mulDivDown(
                         (rewardsIndex[rewardToken] +
-                            claimableAmounts[i].mulDivDown(SCALE, totalSupply())) -
-                            userRewardsData.index,
+                            claimableAmounts[i].mulDivDown(SCALE, supply)) - userRewardsData.index,
```

Comment:
These formulas are really weird, because we scale the claimable amount to 10e36 and mulDiv at the same time.

Reply:
It could be fixed with #185

Reply:
Ok I can do it but this was not the goal of this PR